libgadget/treewalk.c

```diff
@@ -318,7 +321,7 @@ treewalk_build_queue(TreeWalk * tw, int * active_set, const int size, int may_ha
     size_t *nqthr = ta_malloc("nqthr", size_t, All.NumThreads);
     int **thrqueue = ta_malloc("thrqueue", int *, All.NumThreads);
 
-    gadget_setup_thread_arrays(queue, thrqueue, nqthr, size, All.NumThreads);
+    gadget_setup_thread_arrays(queue, thrqueue, nqthr, tsize, All.NumThreads);
 
     /* We enforce schedule static to ensure that each thread executes on contiguous particles.*/
     #pragma omp parallel for schedule(static)
```

Comment:
I would suggest using a for loop in parallel block with explicity static scheduling rather than using the static clause here. Because the cause doesn't say it guarentees the number of iterations per thread is never greater than tsize.

Reply:
intel's manual suggests it is: https://software.intel.com/en-us/articles/openmp-loop-scheduling

Are you just worried that chunk_size may be different? Would it be guaranteed if I manually set chunk_size?